### PR TITLE
Issue 5510: get current

### LINKED ISSUE #5510: Need to add get current for all FSE templates
### Description

Need to add get current for all FSE templates. Single post, page templates, CPT, current Author name, current category. tag, etc

### Other plugins or websites examples

_No response_

### How to implement it

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by elzadj**
# Description

Add get current to DC on various Templates: archives, CPT, Single Posts...

screenshot-wordpress local-2024 03 27-20_59_20

Fixes #5510 

# How Has This Been Tested?

Please check in all places you can think of. Backend and frontend.
Also hides WP 6.5 new Fonts from our DC custom post types.
Also fixes link toolbar UI, at least on 6.5

# Test checklist

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced content retrieval logic for taxonomy, archives, categories, tags, and user content based on specific attributes and conditions.
	- Introduced handling for archive types in dynamic content extensions, including specific processing for archive-type fields.
	- Improved UI components for dynamic content selection and link control with added styling and logic adjustments.

- **Bug Fixes**
	- Fixed a typo in the `UnlimitedCharacterPopover` component name.
	- Adjusted logic to correctly filter and display relation options in the `ContextLoop` component.

- **Refactor**
	- Updated various components and utilities to support new content retrieval logic, including adjustments for handling 'archive' type and template-specific conditions.

- **Style**
	- Added specific styling rules for link control elements in the editor.

### CODE REVIEW

**Review by Olekrut — CHANGES_REQUESTED** [review]
Works good. Frontend at least

Few points:
1) backend does not look good
single post> continer, row with CL get current > create text and enable dc
https://jam.dev/c/ffe46a61-6ca6-4e87-ae90-bb86c6f556b4
2) Should not Archive type be visible only for Archive templates
image

**Comment by elzadj** [comment]
> Works good. Frontend at least
> 
> Few points:
> 
> 1. backend does not look good
>    single post> continer, row with CL get current > create text and enable dc
>    https://jam.dev/c/ffe46a61-6ca6-4e87-ae90-bb86c6f556b4
> 2. Should not Archive type be visible only for Archive templates
>    image

Going to hide both, thanks!

**Comment by elzadj** [comment]
@Olekrut please check visibility of the options now.

**Review by Olekrut — CHANGES_REQUESTED** [review]
1) I see "archive type" not only for Archive templates
image
2) Backend does not show correct value. Actually, should not be value if we are using current. Guess, better if on backend we will see "Title:current" or "Tag:current" , etc
https://jam.dev/c/fe1e0862-9948-400c-897d-0519b5959d37
3) Is it ok, that i see > Post : get current > on Page template?
image
the same in reverse case
image
4) Hard to understand difference between 
type Archive, field Archive type, and when we are using relation Get current archive items, and get current items when we are using type Archive, or field Archive type
And when we should use each one... 
5) not sure, but guess, FSE gets slower on this brunch

**Comment by elzadj** [comment]
1. It also has a type, Single, I vote for leaving it in case people want to add a prefix. Can hide it too.
2. Not sure here, maybe a real example is better, but then we may see 'No content found'. Problem: a placeholder like "Tag:current" is much longer than an actual tag, so it may confuse users with the design. "Title:current" could be much shorter than an actual Title, etc. **Suggestion**: show a real example if it exists, show a placeholder, if it doesn't.
3. Don't see it on Single Posts - Pages, will hide on Pages - Post
4. Please suggest an alternative then. "Archive" is under "Type" menu, so not sure how it's possible to confuse it (it's a WP type). "Archive type" is under another menu, but we can rename it to "Archive's kind" or "Archive type's name". We can rename "Get current archive items" to "Get current archive's posts", "but "item" is a standard WP term when we may show something other than posts. "Get current item" we can rename to "Get current item data" . 
**I summarize my suggestion for renaming**:

"Archive" = "Archive" (in the type menu")
"Archive type" = "Archive type's name" (values are 'category,' 'tag', etc., it's what WP is using)
"Get current archive items" = "Get the current archive's posts"
"Get current item" = "Get the current item's data"

**Comment by kyrapieterse** [comment]
@Olekrut MVP

**Comment by Olekrut** [comment]
@elzadj ok, agree, only about  point 2, 
when we are using current for FSE, there should be like Tag:current, because we cant have there static DC value, because it is template

**Comment by elzadj** [comment]
> @elzadj ok, agree, only about point 2, when we are using current for FSE, there should be like Tag:current, because we cant have there static DC value, because it is template

And if they want to show something like "Tag: business" on frontend? Like we do in the theme templates we want to add to the theme.
Then on the backend it's going to looks like "Tag: Tag: current"

**Comment by Olekrut** [comment]
> > @elzadj ok, agree, only about point 2, when we are using current for FSE, there should be like Tag:current, because we cant have there static DC value, because it is template
> 
> And if they want to show something like "Tag: business" on frontend? Like we do in the theme templates we want to add to the theme. Then on the backend it's going to looks like "Tag: Tag: current"

Yes, you are right, but backend is weird anyway.

**Comment by Olekrut** [comment]
@elzadj ok, with previous, new questions:
1) https://jam.dev/c/da6c052a-5aae-4687-bad3-82f33b34cc9e
create post > create text inside > set dc current > Error and field validation
2)https://jam.dev/c/6747670b-3ba2-4c68-bcdb-37b52739eaf3
 cant enable DC link for current on FSE, open single post/page template > create dc current text block > try to enable dc link

**Comment by Olekrut** [comment]
1) first point is no fixed
https://jam.dev/c/a0e44d73-de19-4380-b431-1299f70298c2

**Review by Olekrut — CHANGES_REQUESTED** [review]
last 2 points are not fixed
https://jam.dev/c/a0e44d73-de19-4380-b431-1299f70298c2
and new ones:

3) https://jam.dev/c/486c28f0-caee-48e4-8674-9e50089a8812
cant to make it with our blocks, frontend does not work
image
4) type-archive is available only for All archive template, > not for Author archive template
image
https://jam.dev/c/6dccce07-8b9b-4c60-b496-3bddee0af82b
image

**Comment by elzadj** [comment]
> last 2 points are not fixed

@Olekrut I can't recreate both, works for me. Try on a new page, if you still see it - give me the blocks' code please.

**Comment by elzadj** [comment]
3. Fixing author.
4. You need to choose the proper archive. Author for authors, Tag for Tag. I propose to rename 'Archive' to 'All archives' there to not confuse users.

**Review by Olekrut — CHANGES_REQUESTED** [review]
1) Should we to make the same for WooCommerce in this issue, or we will not have it in our theme, and we can make it later?
image
2) type is empty if duplicate dc block
https://jam.dev/c/1a8ca976-927f-4350-87bd-8bf04273ab18
3) Looks like links are not correct if enable it for archive template
https://jam.dev/c/9f335432-9278-48ed-95d4-dbaf0ae4652b
image

4) Guess, problem with validation is not fixed. Noticed for All archives template
https://jam.dev/c/98d62871-dfa1-478a-a230-fd49db2bebbf
5) again see it, all archives on Author archive template
https://jam.dev/c/6649b2f7-48c6-418c-a2d8-23a1886db393

File: src/components/dynamic-content/index.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
 	useMemo,
 	useState,
 } from '@wordpress/element';
-import { resolveSelect, useSelect } from '@wordpress/data';
+import { resolveSelect, useSelect, select } from '@wordpress/data';
```
[the remark this answers is not in the document]
Comment (elzadj) [reply src/components/dynamic-content/index.js:12]: @coderabbit we use select on line 251 in the same file

**Comment by elzadj** [comment]
1. Next week 
2. Everything else is ready for tests.

**Comment by Olekrut** [comment]
author link is always the author with id 1
image

**Comment by elzadj** [comment]
> author link is always the author with id 1 image

Same on master for author by id. Fixing in this one.

**Review by Olekrut — APPROVED** [review]
Well done, Sv! 🥇 
left woo

**Pull request merged**